### DISCUSSION TITLE
Reduced GB Network

### DISCUSSION THREAD

**Discussion by ccookeopen** [bukhsh/oats#111] [Q&A]
Hi,

Many thanks for providing the reduced GB network model. I have a couple of questions:

- for a number of generators, the maximum and minimum real power outputs are the same.
- Might it be possible to indicate the intended fuel type for each generator, with a view to extending the model to cater for dynamic simulations?

Thanks, C.

**Comment by bukhsh** [comment]
Many thanks for the questions, Christian.

The max and min outputs should not be the same. I have updated the data set on GitHub and have changed the PGLB (lower bound of generators) to zero for all generators. The reason I have changed PGLB to zero is to ensure feasibility for DCOPF/ACOPF models. As you know, thermal generators have a non-zero minimum stable operating point. You can find that information in the literature for each generation type. I have also updated the 'fuel type' of generators in the data. Please note that this is rather an old data set (2015ish) so the generation mix in GB has changed since - especially the coal generation which is on its way out. Please take a look at National Grids ETYS for more information on generation mix in the GB. Als please feel free to come back to me if you have any question.

Cheers
Waqquas

**Reply by ccookeopen** [reply]
That's great, Waqquas. Many thanks for the reply.

I wondered whether the "owner" field of the buses can be used to distribute them geographically. A number of them are obviously so, but a key would be useful. Within the EYTS boundaries, for instance.

https://www.nationalgrideso.com/document/181266/download

**Reply by bukhsh** [reply]
The bus names are easy enough to identify their geographical location e.g. 
HARK -> Harker, 
STEW -> Stella West

You can also refer to Appendix B of the National Grid's Electricity Ten Year Statement for more information about the node names and their location. The information is divided into three transmission owners in GB.

I understand that your work is investigating inertia in the GB grid...how important is the network for your methods to work? By network, I mean transmission characteristics (r,x and b values) of the transmission lines. Can you work with just transfer limits on the boundaries? If so then maybe a better representation of GB would be using the 14 Grid Supply Point (GSP) Groups.

**Reply by bukhsh** [reply]
FYI - The network diagram with geographical location of the nodes is here:

https://github.com/bukhsh/oats/blob/master/docs/pics/GBdiagram.jpg

**Comment by ccookeopen** [comment]
Dear Waqquas,

Many thanks for your assistance. It has been a great help to me.

Best regards,

Christian
________________________________

**Reply by ccookeopen** [reply]
Hi,

I get a "list index out of range" for the timeseries sheet when using GB_ReducedNetwork.xlsx as a test case in oats. 

Any help appreciated.

Christian

**Reply by ccookeopen** [reply]
I resolved this by adding a zero the the second row of the empty timeseries sheet, as well as adding a "storage" sheet containing a "stat" column with a zero.

Can these be include in the sheet without issues, or should the exceptions be caught?

**Reply by bukhsh** [reply]
Hi,

Thanks for this. 

The recent version of OATS contains exceptions on 'storage' and 'timeseries'. I'll extend these exceptions to other sheets as well.

Cheers
Waqquas

**Comment by ccookeopen** [comment]
Hi,

Oats will usually find a solution, though not necessarily within the specified upper and lower bounds. For instance, running an ac load flow on the GB reduced network will give a result that has bus voltages and generator reactive power levels exceeding the given boundaries. Can Oats be used to find how a system can be modified so that these limits are kept?

Thanks,

C

**Reply by bukhsh** [reply]
Hi,

AC load flow problem is not an optimisation problem and therefore will not necessarily respect the upper and lower bounds. See the following report for details on how it is modelled in oats

https://pureportal.strath.ac.uk/en/publications/on-solving-the-load-flow-problem-as-an-optimization-problem

If you want a solution to respect the bounds on variables, then solve an optimal power flow - which is an optimisation problem and the solution will provide an optimal result that respect the bounds.

Waqquas

**Reply by ccookeopen** [reply]
Waqquas,

My understanding is that in optimal power flow the generator active power is to be kept constant, whereas acopf modifies set generator active power significantly. Am I mistaken?

Thanks,

Christian

**Reply by bukhsh** [reply]
Hi Christian,

The optimal power flow problem is an optimisation problem and the objective of a traditional OPF is to minimise the cost of generation to meet a given demand while meeting constraints on generation, network and bus voltages. The objective of OPF is to determine a least-cost dispatch of generation.

However, the load flow problem (which is not an optimisation problem, but a problem of finding a solution of a set of nonlinear equations) the generator set-points are fixed, except for one generator which is called a slack generator. I have written a small subsection in the following article that describes the relationship between the load flow and the OPF problem

https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=6581918

Waqquas

**Reply by ccookeopen** [reply]
Thanks for that, Waqquas.

You might check a bug I've encountered. In the generator tab the "busname" and "name" fields appear to get swapped in the results.xlsx file.

Christian

[I filed a bug]

**Reply by bukhsh** [reply]
Many thanks for this, Christian. I'll fix it in the next iteration.

**Reply by ccookeopen** [reply]
> AC load flow problem is not an optimisation problem and therefore will not necessarily respect the upper and lower bounds. See the following report for details on how it is modelled in oats
> 
> https://pureportal.strath.ac.uk/en/publications/on-solving-the-load-flow-problem-as-an-optimization-problem
> 
> If you want a solution to respect the bounds on variables, then solve an optimal power flow - which is an optimisation problem and the solution will provide an optimal result that respect the bounds.
> 
> Waqquas

I have a modified reduced GB grid model, with a different generation active power profile and scaled load to match it. Running aclf it returns voltage angles of c. -90, voltage magnitudes 1.5+ and generation reactive powers far in excess of limits.

If the line resistances are zeroed, the same model returns small voltage angles, near-unit voltages and reasonable reactive powers. Eigenvalue analysis of the resulting system indicates it is stable. Considering that linearised load flow neglects line losses altogether, is it expected that they should have such a large effect on the aclf result?

**Reply by bukhsh** [reply]
Making line resistances (the r-value in the data) zero will not result in a linear problem. You'll still have sine and cosine functions plus product of bilinear voltages at either end of a line in the nonlinear Kirchhoff's voltage laws.

DCOPF is a linear problem. The assumption in DCOPF is the resistance is zero and the angle differences are small such that sine of angle difference can be approximated by the angle difference itself and cosine of angle differences is zero. This results in a linear problem. It is a linearisation of the ACOPF problem, not a relaxation. 

You can solve ACOPF, which would allow you to have bounds on the voltages as well as active power dispatch of generation. You can fix any number of generators if you like by making PGLB=PGUB in the data set.

**Reply by ccookeopen** [reply]
Dear Waqquas,

I appreciate your patience in answering these queries.

> You can solve ACOPF, which would allow you to have bounds on the voltages as well as active power dispatch of generation. You can fix any number of generators if you like by making PGLB=PGUB in the data set.

This results in a system that doesn't converge. This is a system that passes power flow in Andes. There doesn't appear to be any indication how it might be modified so that it's feasible.

**Reply by bukhsh** [reply]
It's worth trying lower bounds 'pG-\epsilon' and upper bounds 'pG+\epsilon', where \epsilon is a small number. Forcing exact bounds sometimes cause infeasibility, similarly for any voltages that you are fixing.

**Comment by ccookeopen** [comment]
Hi Waqquas,

Can you say how the load is distributed in this and other Oats testcases? Is it based on population for the GB grid?

Thanks,

Christian

**Reply by bukhsh** [reply]
Hi Christian, 
The load data is based on a historic snapshot. At the time (from memory it's from a 2009 report, so quite an old data set!) it would have been the transmission system demand i.e. seen at the Grid Supply Level. As you know, the demand for electricity varies throughout a day and it does not vary proportionally. Furthermore, the patterns of electricity consumption are changing because of embedded generation etc so it'd be useful to review and revise the demand data within the testcase. 

Cheers
Waqquas
